cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -193,7 +194,9 @@ protected void startUp() throws Exception {
             Constants.Service.PREVIEW_HTTP));
     logSubscriberService = previewInjector.getInstance(PreviewTMSLogSubscriber.class);
     logSubscriberService.startAndWait();
-    dataSubscriberService = previewInjector.getInstance(PreviewDataSubscriberService.class);
+    TransactionRunner transactionRunner = previewInjector.getInstance(TransactionRunner.class);
+    dataSubscriberService = new PreviewDataSubscriberService(previewCConf, messagingService,
```

Comment:
Use previewInjector to getInstance of `PreviewDataSubscriberService` instead of using the constructor.